app/Controllers/ValidateFeeController.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\App\Controllers;
+
+use InvalidArgumentException;
+use Silex\Application;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use WMDE\Euro\Euro;
+use WMDE\Fundraising\MembershipContext\UseCases\ValidateMembershipFee\ValidateFeeRequest;
+use WMDE\Fundraising\MembershipContext\UseCases\ValidateMembershipFee\ValidateMembershipFeeUseCase;
+
+class ValidateFeeController {
+
+	private $app;
+	private $httpRequest;
+
+	public function __construct( Application $app, Request $httpRequest ) {
+		$this->app = $app;
+		$this->httpRequest = $httpRequest;
+	}
+
+	public function validateFee(): Response {
+		try {
+			$fee = $this->euroFromRequest();
+		}
+		catch ( InvalidArgumentException $ex ) {
+			return $this->newJsonErrorResponse( 'not-money' );
+		}
+
+		$request = ValidateFeeRequest::newInstance()
+			->withFee( $fee )
+			->withInterval( (int)$this->httpRequest->request->get( 'paymentIntervalInMonths', '0' ) )
+			->withApplicantType( $this->httpRequest->request->get( 'addressType', '' ) );
+
+		$response = ( new ValidateMembershipFeeUseCase() )->validate( $request );
+
+		if ( $response->isSuccessful() ) {
+			return $this->app->json( [ 'status' => 'OK' ] );
```

Comment:
Since `app` is on its way out for the Silex -> Symfony transition, please initialize the `JsonResponse` directly. See `ValidateDonorController` for an example.

Reply:
Hah. That seems better even if you are not migrating away from Silex.